src/Core/Animation/KeyFramedValue.hpp

```diff
@@ -0,0 +1,215 @@
+#ifndef RADIUMENGINE_KEYFRAMEDVALUE_HPP
+#define RADIUMENGINE_KEYFRAMEDVALUE_HPP
+
+#include <map>
+#include <set>
+
+#include <Core/Containers/AlignedAllocator.hpp>
+#include <Core/Math/Math.hpp>
+#include <Core/Types.hpp>
+
+namespace Ra {
+namespace Core {
+namespace Animation {
+
+/**
+ * KeyFramedValueBase defines the following concept:
+ * any variable (e.g. transformation matrix, color, scalar value, flag) that
+ * needs to be animated can be controlled by a `KeyFramedValue`, which inherits
+ * KeyFramedValueBase (see dedicated documentation).
+ *
+ * KeyFramedValueBase defines the basic interface available to control keyframes:
+ * insertKeyFrame, removeKeyFrame and moveKeyFrame.
+ */
+class RA_CORE_API KeyFramedValueBase
+{
+  public:
+    KeyFramedValueBase() {}
+
+    virtual ~KeyFramedValueBase() {}
+
+    /**
+     * \returns the number of keyframes.
+     */
+    virtual size_t size() const = 0;
+
+    /**
+     * \returns the ordered list of the points in time where a keyframe is defined.
+     */
+    virtual inline std::vector<Scalar> getTimes() const = 0;
```

Comment:
well this was an helper function, but it could be removed

Reply:
Keeping this function could introduce some data incoherency between usage of the KeyFramedValues. Imagine you get a copy to, e.g. build an interface but refine only the original data making the interface to diverge from the original data. This will not be robust.
And if it is a helper function, it is then not mandatory. So, it should'nt be pure virtual. And as it can't be something else, remove it and let the user build its own helpers function.

Reply:
this method was needed to get all the time points where a frame is defined without having to cast from the base class (which the ui container is built on) to the derived class (which is templated w.r.t. the `VALUE_TYPE`)